### PR TITLE
Light mode: sync status ready if percentage > 0.99

### PR DESCRIPTION
**Pull request by Unisay**
- [x] Instead of returning sync status "Syncing 100%" wallet returns status "Ready" when percentage > 99%. This way status doesn't blink from `Ready` to `Syncing` when a new latest block is available but isn't fetched yet.

### Comments

In the node mode the wallet syncs the whole blockchain (block by block), so `syncProgress` function returns a percentage of the blockchain fetched so far.

In the light mode wallet syncs only the wallet history (only the blocks that are relevant for the specific wallet) and it doesn't make sense to return a sync progress as a percentage of the blockchain, but as a percentage of the history. 

This means that `NetworkLayer.syncProgress :: SlotNo -> Percentage` function simply doesn't make sense for the light mode. This PR doesn't provide a better abstraction but makes the current one slightly less wrong.

### Issue Number

ADP-1508

### CODE REVIEW

File: lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs [hunk 1]
```diff
@@ -401,8 +402,11 @@ withNetworkLayer tr network np project k = do
         BF.Block {_blockSlot} <- bfGetLatestBlock bfLayer
         let latestSlot = maybe 0 BF.unSlot _blockSlot
             currentSlot = fromIntegral (unSlotNo s)
-            percentage = currentSlot % latestSlot
-        pure $ Syncing $ Quantity $ clipToPercentage percentage
+            percentage = clipToPercentage $ currentSlot % latestSlot
+        pure $
+            if percentageToDouble percentage > 0.99
```
Comment (HeinrichApfelmus) [inline lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs:407]: By the way, for numerical algorithms, there is a rule of thumb which says "Never compare `Double`!", as this usually leads to numerical instability and weird artifacts. (For example, GHCi will evaluate `3*0.1 > 0.3` to `True`.)

The less extreme variant of this rule is to "Always double check comparison of `Double`". I have double-checked this code and I believe that it is perfectly fine here. 🤓

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good to me! 😊 Side remark on comparisons of `Double`.

File: lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs (hunk 1 above)
Comment (Unisay) [reply lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs:407]: Yeah, I think we're good in this particular case as there is no much arithmetic to loose precision and we can tolerate some imprecision as there is no `==` operator in use. 
https://www.geeksforgeeks.org/problem-in-comparing-floating-point-numbers-and-how-to-compare-them-correctly/

**Comment by Unisay** [comment]
bors retry

**Pull request merged**